code/game/turfs/turf.dm

```diff
@@ -369,4 +371,24 @@
 	return 1
 
 /turf/proc/can_lay_cable()
-	return can_have_cabling() & !intact
\ No newline at end of file
+	return can_have_cabling() & !intact
+	
+/turf/proc/addSmell(var/smellName, var/qty)
+	if(istype(src, /turf/simulated))
```

Comment:
Indeed, you can create an empty proc for `/turf`, and a full-fledged proc for `/turf/simulated`